module/vision/GoalDetector/src/GoalDetector.cpp

```diff
@@ -235,7 +236,31 @@ namespace module::vision {
                             g.screen_angular = cartesianToSpherical(g.post.bottom).tail<2>();
                             g.angular_size   = Eigen::Vector2f::Constant(std::acos(radius));
 
-                            goals->goals.push_back(std::move(g));
+                            /***********************************************
+                             *                  THROWOUTS                  *
+                             ***********************************************/
+
+                            if (config.debug) {
+                                log<NUClear::DEBUG>("**************************************************");
+                                log<NUClear::DEBUG>("*                    THROWOUTS                   *");
+                                log<NUClear::DEBUG>("**************************************************");
+                            }
+                            bool keep = true;  // if false then we will not consider this as a valid goal post
+
+                            // If the goal is too far away, get rid of it!
+                            if (distance > config.max_goal_distance) {
+                                keep = false;
+                                log<NUClear::DEBUG>(
+                                    fmt::format("Goal discarded: goal distance ({}) > maximum_goal_distance ({})",
+                                                distance,
+                                                config.max_goal_distance));
+                                log<NUClear::DEBUG>("--------------------------------------------------");
+                            }
```

Comment:
Why would the robot possibly pick up a goal to far away in the first place?

Reply:
Goals on other fields, or things that may look like goals but aren't

Reply:
Basically vision isn't perfect.